### PR TITLE
Implementing UIA supporting for Form control

### PR DESCRIPTION
**Pull request by vladimir-krestov**
## Issue description
- Form doesn't support UIA so it blocks #2074 because Narrator can't announce tooltip text if no controls on a form don't support UIA provider.

## Proposed changes
- Set SupportsUiaProviders property as true
- Use FormAccessibleObject as AccessibleObject instead default

**This allows customizing Form AccessibleObject how we want. 
For example, fix UIA navigation in the accessibility tree. It changes the focus order of Narrator. 
Form AccessibleObject works as before.**

## Customer Impact

- <del>Improved Bounds</del> No

## Regression? 

- No

## Risk

- No

## Test methodology 

- Manual UI testing

## Accessibility testing  
- Using Inspect tool

 

## Test environment(s) 
- .NET Core Version: 3.1.0-preview1.19458.7
- Microsoft Windows [Version 10.0.18362.356]

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2096)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 1]
```diff
@@ -7288,6 +7295,62 @@ protected override void WndProc(ref Message m)
             }
         }
 
+        /// <summary>
+        ///  Form control accessible object with UI Automation provider functionality.
+        ///  This inherits from the base ControlAccessibleObject
+        ///  to have all base functionality.
+        /// </summary>
+        internal class FormAccessibleObject : ControlAccessibleObject
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: Can this be placed in a separate file? e.g. `Form.FormAccessibleObject.cs`

Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: Yes, I'll change this file location.

Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: This file location is changed.

**Comment by vladimir-krestov** [comment]
Needs review and approve before testing.

**Comment by RussKie** [comment]
Please don't forget to squash

**Comment by vladimir-krestov** [comment]
I'll squash commits after testing before final approval. Thanks for reminding.

**Comment by vladimir-krestov** [comment]
Removed Bounds implementation because it breaks RT test cases. At the moment Form AccessibleObject the same as before, but supports UIA provider.

**Comment by merriemcgaw** [comment]
business case approved. Please add ask-mode when ready with the tested fix.

**Comment by vladimir-krestov** [comment]
Testers approved the fix. :heavy_check_mark:
But Bounds property and BoundingRectangle property don't match.
BoundingRectangle is the correct yellow rectangle of Inspect.
Bounds made as before the fix. All automation tests passed.

**Pull request merged**